manim/mobject/coordinate_systems.py

```diff
@@ -607,7 +607,20 @@ def get_riemann_rectangles(
 
         # setting up x_range, overwrite user's third input
         if x_range is None:
-            x_range = self.x_range
+            if bounded_graph is None:
+                x_range = [graph.t_min, graph.t_max, graph.t_step]
+            else:
+                x_min = None
+                x_max = None
+                if graph.t_min > bounded_graph.t_min:
+                    x_min = graph.t_min
+                else:
+                    x_min = bounded_graph.t_min
+                if graph.t_max < bounded_graph.t_max:
+                    x_max = graph.t_max
+                else:
+                    x_max = bounded_graph.t_max
+                x_range = [x_min, x_max, graph.t_step]
```

Comment:
x_min=min(bounded.t_min, graph.t_min) and x_max=max(bounded.t_max, graph.t_max) should make this simpler